Update SDL-0219 with accepted revisions (#694)
smartdevicelink/sdl_evolution: proposals/0219-ios-check-type.md

## Proposed solution

Mark method as private:
`-(nullable id)sdl_objectForName:(SDLName)name;`. 

Instead of method above use next methods:

> For optional values, can be used functions that return object of classType or nil:

Reviewer: ```suggestion
For optional values, can use functions that return object of classType or nil:
```

`- (nullable id)sdl_objectForName:(SDLName)name ofClass:(Class)classType;`.
`- (nullable NSArray *)sdl_objectsForName:(SDLName)name ofClass:(Class)classType`.

> For mandatory values, next functions try to return object of classType overwise error would be filled. 

Reviewer: ```suggestion
For mandatory values, the following functions try to return object of classType overwise error will be filed. 
```

> In case, when error isn't `nil`, returned object can be not type of `classType`:

Reviewer: ```suggestion
When the object is not of type `classType`, the error will not be `nil`
```

`- (nullable id)sdl_objectForName:(SDLName)name ofClass:(Class)classType error:(NSError **)error`
`- (nullable NSArray *)sdl_objectsForName:(SDLName)name ofClass:(Class)classType error:(NSError **)error;`.

For suitable working with SDLEnums create the following methods:

> `- (nullable SDLEnum)sdl_enumForName:(SDLName)name;`

Reviewer: Is this needed? it's not included in the requested revisions.

Author: In my view, we can save this method if we need to getting SDLEnum and don't want to handle error.
```
- (nullable SDLEnum)sdl_enumForName:(SDLName)name {
    return [self sdl_objectForName:name ofClass:NSString.class];
}
- (nullable id)sdl_objectForName:(SDLName)name ofClass:(Class)classType {
    NSError *error;
    id object = [self sdl_objectForName:name ofClass:classType error:&error];
    return error == nil ? object : nil;
}
```

> `- (nullable SDLEnum)sdl_enumForName:(SDLName)name error:(NSError *);`.

Reviewer: ```suggestion
`- (nullable SDLEnum)sdl_enumForName:(SDLName)name error:(NSError **)error;
```

> `- (nullable NSArray<SDLEnum> *)sdl_enumsForName:(SDLName)name;`

Reviewer: This can be removed, per the requested revisions.

Author: This method calls inside method with error, like example above

> `- (nullable NSArray<SDLEnum> *)sdl_enumsForName:(SDLName)name error:(NSError *);`.

Reviewer: ```suggestion
`- (nullable NSArray<SDLEnum> *)sdl_enumsForName:(SDLName)name error:(NSError **)error;
```

## Potential downsides

> Store can return wrong type that would be highlighted by error.

Reviewer: ```suggestion
Store can return wrong type, which would be highlighted by an error.
```
